task/s2i/0.2/README.md

```diff
@@ -0,0 +1,144 @@
+
+- **IMAGE_NAME**: Reference of the image S2I will produce.
+- **BUILDER_IMAGE**: The location of the s2i builder image.
+- **PATH_CONTEXT**: Source path from where s2i command need to be run
+  (_default: ._).
+- **TLSVERIFY**: Verify the TLS on the registry endpoint (for push/pull to a
+  non-TLS registry) (_default:_ `true`)
+
+## Workspaces
+
+- **source**: A [Workspace](https://github.com/tektoncd/pipeline/blob/master/docs/workspaces.md) containing the source to build.
+- **sslcertdir**: An [_optional_ Workspace](https://github.com/tektoncd/pipeline/blob/v0.17.0/docs/workspaces.md#optional-workspaces) containing your custom SSL certificates to connect to the registry. Buildah will look for files ending with \*.crt, \*.cert, \*.key into this workspace. See [this sample](./samples/openshift-internal-registry.yaml) for a complete example on how to use it with OpenShift internal registry.
+
+## Results
+
+- **IMAGE_DIGEST**: Digest of the image just built.
+
+## ServiceAccount
+
+S2I builds an image and pushes it to the destination registry which is
+defined as a parameter. In order to properly authenticate to the
+remote container registry, it needs to have the proper
+credentials. The credentials can be provided through a service
+account. See
+[Authentication](https://github.com/tektoncd/pipeline/blob/master/docs/auth.md#basic-authentication-docker)
+for further details.
+
+If you are running on OpenShift, you also need to allow the service
+account to run privileged containers because OpenShift does not allow
+containers run as privileged containers by default unless explicitly
+configured, due to security considerations.
+
+Run the following in order to create a service account named
+`pipeline` on OpenShift and allow it to run privileged containers:
+
+```bash
+oc create serviceaccount pipeline
+oc adm policy add-scc-to-user privileged -z pipeline
+oc adm policy add-role-to-user edit -z pipeline
+```
+
+## Usage
+
+This PipelineRun runs the Task to fetch a Git repo, and build and push a
+container image using s2i and a nodejs builder image.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: s2i-test-pipeline-run
+spec:
+  pipelineSpec:
+    workspaces:
+      - name: shared-workspace
+      - name: sslcertdir
+        optional: true
+    tasks:
+      - name: fetch-repository
+        taskRef:
+          name: git-clone
+        workspaces:
+          - name: output
+            workspace: shared-workspace
+        params:
+          - name: url
+            value: https://github.com/sclorg/nodejs-ex
+          - name: subdirectory
+            value: ""
+          - name: deleteExisting
+            value: "true"
+      - name: s2i
+        taskRef:
+          name: s2i
+        runAfter:
+          - fetch-repository
+        workspaces:
+          - name: source
+            workspace: shared-workspace
+        params:
+          - name: BUILDER_IMAGE
+            value: docker.io/centos/nodejs-8-centos7
```

Comment:
isnt there a quay based version?

Reply:
Added the quay based image in the example and also added in the note to use quay based image if somebody wants to